src/test/java/ScheduleTest.java

```diff
@@ -1,59 +1,60 @@
-        endTime1 = LocalTime.of(10, 0, 0);
-        startTime2 = LocalTime.of(16, 0, 0);
-        endTime2 = LocalTime.of(17, 0, 0);
-        startTime3 = LocalTime.of(13, 0, 0);
-        endTime3 = LocalTime.of(14, 0, 0);
+        LocalTime startTime1 = LocalTime.of(9, 0, 0);
+        LocalTime endTime1 = LocalTime.of(10, 0, 0);
+        LocalTime startTime2 = LocalTime.of(16, 0, 0);
+        LocalTime endTime2 = LocalTime.of(17, 0, 0);
+        LocalTime startTime3 = LocalTime.of(13, 0, 0);
+        LocalTime endTime3 = LocalTime.of(14, 0, 0);
 
-        day = DayOfWeek.MONDAY;
+        String lectureLocation = "SS 2135";
+        String tutorialLocation = "MP203";
+        DayOfWeek day = DayOfWeek.MONDAY;
 
-        lec1 = new Section();
+        Timeslot lec1 = new Timeslot(startTime1, endTime1, day, lectureLocation);
+        Timeslot lec2 = new Timeslot(startTime2, endTime2, day, lectureLocation);
+        Timeslot lec3 = new Timeslot(startTime3, endTime3, day, lectureLocation);
 
-        lec1 = new Section.Builder("LEC-0101").inRoom("SS 2135").startsAt(startTime1).endsAt(endTime1).onDay(day)
-                .build();
-        lec2 = new Section.Builder("LEC-0201").inRoom("SS 2135").startsAt(startTime2).endsAt(endTime2).onDay(day)
-                .build();
-        lec3 = new Section.Builder("LEC-0301").inRoom("SS 2135").startsAt(startTime3).endsAt(endTime3).onDay(day)
-                .build();
+        Timeslot tut1 = new Timeslot(startTime1, endTime1, day, tutorialLocation);
+        Timeslot tut2 = new Timeslot(startTime2, endTime2, day, tutorialLocation);
+        Timeslot tut3 = new Timeslot(startTime3, endTime3, day, tutorialLocation);
 
-        tut1 = new Section.Builder("TUT-0101").inRoom("MP203").startsAt(startTime1).endsAt(endTime1).onDay(day).build();
-        tut2 = new Section.Builder("TUT-0201").inRoom("MP203").startsAt(startTime2).endsAt(endTime2).onDay(day).build();
-        tut3 = new Section.Builder("TUT-0301").inRoom("MP203").startsAt(startTime3).endsAt(endTime3).onDay(day).build();
+        lec1Section = new Section("TST101 LEC0101 F", List.of(lec1));
+        lec2Section = new Section("TST101 LEC0201 F", List.of(lec2));
+        lec3Section = new Section("TST101 LEC0301 F", List.of(lec3));
 
-        lecsToAdd = new ArrayList<>(List.of(lec1, lec2, lec3));
-        tutsToAdd = new ArrayList<>(List.of(tut1, tut2, tut3));
+        tut1Section = new Section("TST101 TUT0202 F", List.of(tut1));
+        tut2Section = new Section("TST202 TUT0202 F", List.of(tut2));
+        tut3Section = new Section("TST303 TUT0202 F", List.of(tut3));
+
+        lecsToAdd = List.of(lec1Section, lec2Section, lec3Section);
+        tutsToAdd = List.of(tut1Section, tut2Section, tut3Section);
     }
 
     @Test(timeout = 50)
     public void testAddLectures() {
+        Schedule schedule = new Schedule();
 
-        ArrayList<Section> expected = new ArrayList<>(List.of(lec1, lec3, lec2));
+        ArrayList<Section> expected =
```

Comment:
Should this also be a List? Unless there is a reason for keeping it as an ArrayList. I've commented on other instances of this, so disregard those for the same reason as this one.

Reply:
yeah. I think that'll be a separate PR